src/Orleans.Core/Configuration/OptionLogger/DefaultOptionsFormatter.cs

```diff
@@ -30,21 +32,60 @@ public IEnumerable<string> Format()
             return typeof(T)
                 .GetProperties()
                 .Where(prop => prop.GetGetMethod() != null && prop.GetSetMethod() != null)
-                .Select(FormatProperty)
+                .SelectMany(FormatProperty)
                 .ToList();
         }
 
-        private string FormatProperty(PropertyInfo property)
+        private IEnumerable<string> FormatProperty(PropertyInfo property)
         {
+            var result = new List<string>();
             var name = property.Name;
             var value = property.GetValue(this.options);
             var redactAttribute = property.GetCustomAttribute<RedactAttribute>(inherit: true);
 
-            return OptionFormattingUtilities.Format(
-                name, 
-                redactAttribute != null 
-                    ? redactAttribute.Redact(value)
-                    : value);
+            // If it is a dictionary -> one line per item
+            if (typeof(IDictionary).IsInstanceOfType(value))
```

Comment:
I think we need to check to see if the property has a redact action, and if so, redact it, and only check specific data types if not redacted.  The reason is that the redact action takes and object so it's intuitive that the redact would be called on the entire property, and not on parts of the property we select by some internal logic.

Reply:
I agree